examples/40_advanced/example_metrics.py

```diff
@@ -186,3 +193,43 @@ def error_wk(solution, prediction, extra_argument):
 predictions = cls.predict(X_test)
 score = error_rate(y_test, predictions)
 print(f"Error score {score:.3f} using {error_rate.name:s}")
+
+
+#############################################################################
+# Sixth example: Use a metric with additional argument which also needs xdata
+# ===============================================================
+"""
+Finally, *Auto-sklearn* also support metric that require the train data (aka x_data) to
+compute a value. This can be useful if one only cares about the score on a subset of the
+data.
+"""
+
+accuracy_scorer = autosklearn.metrics.make_scorer(
+    name="accu_X",
+    score_func=metric_which_needs_x,
+    optimum=1,
+    greater_is_better=True,
+    needs_proba=False,
+    needs_x=True,
+    needs_threshold=False,
+    consider_col=1,
+    threshold=20,
+)
+cls = autosklearn.classification.AutoSklearnClassifier(
+    time_left_for_this_task=60,
+    per_run_time_limit=30,
+    seed=1,
+    metric=accuracy_scorer,
+    ensemble_size=0,
+)
+cls.fit(X_train, y_train)
+
+predictions = cls.predict(X_test)
+score = metric_which_needs_x(
+    y_test,
+    predictions,
+    x_data=X_test,
+    consider_col=1,
+    threshold=18.8,
```

Comment:
Why do you change the threshold here?

Reply:
sorry, this should be an extra argument to the metric. I renamed it

Reply:
Do you then also need to change the variable name in the definition of `accuracy_scorer` above:

```
accuracy_scorer = autosklearn.metrics.make_scorer(
    name="accu_X",
    score_func=metric_which_needs_x,
    optimum=1,
    greater_is_better=True,
    needs_proba=False,
    needs_X=True,
    needs_threshold=False,
    consider_col=1,
    threshold=20,
)
```

Reply:
True. Done.